app/lib/shared/task_scheduler.dart

```diff
@@ -14,60 +14,70 @@ typedef Future TaskRunner(Task task);
 
 // ignore: one_member_abstracts
 abstract class TaskSource {
+  final StreamController<Task> _controller = new StreamController();
+
   /// Returns a stream of currently available tasks at the time of the call.
-  Stream<Task> currentTasks();
+  Stream<Task> start() => _controller.stream;
+
+  void addTask(Task task) {
+    _controller.add(task);
+  }
 }
 
 /// Tasks coming from through the isolate's receivePort, originating from a
 /// HTTP handler that received a ping after a new upload.
-class ManualTriggerTaskSource implements TaskSource {
-  final Stream taskReceivePort;
-  final int capacity;
-  StreamSubscription _subscription;
+class ManualTriggerTaskSource extends TaskSource {
   final Set<Task> _triggered = new Set();
+  StreamSubscription<Task> _subscription;
+  Timer _timer;
 
-  ManualTriggerTaskSource(this.taskReceivePort, {this.capacity: 100}) {
+  ManualTriggerTaskSource(Stream taskReceivePort, {int capacity: 100}) {
     _subscription = taskReceivePort.listen((Task task) {
       // protect against spamming the manual trigger
       if (_triggered.length < capacity) {
         _triggered.add(task);
       }
+      if (_timer == null) {
+        _timer = new Timer(const Duration(minutes: 1), () {
+          _timer = null;
+          for (Task task in _triggered) {
+            _controller.add(task);
+          }
+          _triggered.clear();
+        });
+      }
```

Comment:
This would also look **much** nicer if we just used an `async*` generator with a sleep in the for loop body :-/